### PR TITLE
Feature/36012 - Update slider module

### LINKED ISSUE #179: Update slider module
Actually, the slide content is set to product (element product grid)
Content slide should be set in config (as other component) and shouldn't be related with product element/module

### PR DESCRIPTION
**Pull request by kjugi**
+ Delete single product preview in elements, instead of this there is rendering done elements
+ Cleanup styles and config

Fixes #179 
https://dev.snowdog.pro/issues/36012

### CODE REVIEW

File: components/03-modules/slider/slider.hbs [hunk 1]
```diff
@@ -1,37 +1,8 @@
...
-                        {{#if badge}}
-                            {{ render '@badge' this.badge }}
-                        {{/if}}
-                    </a>
-                </div>
+            {{#each elements }}
+                {{ render (component content) contentContext }}
             {{/each}}
         </div>
     </div>
```
Comment (anqaka) [inline components/03-modules/slider/slider.hbs]: code looks ok for me but the slider on wider screen doesn't work properly, I see mutliple slides and I cannot see the arrows. It should work like that? we should be able to set how many slides is visible? and there are some option to media query? I try also with images as content. Could you please recheck this module with multiple slides elements?

Comment (kjugi) [reply components/03-modules/slider/slider.hbs]: ok, for now, I've added new div which represents one item in the slider. We can add class to it and modify as we want with size. Navigation in the slider, for now, can have own class and we can delete hiding arrow on the desktop (b/c in the project are hidden in desktop). For the preview, I've added images elements. And to manage configu of slider we can pass attributes as `data` and give there props as in docs. You can see in the preview.

File: components/03-modules/slider/_slider.scss [hunk 2]
```diff
@@ -63,83 +44,6 @@ $slider__icon-fill-hover           : darken($color-primary, 10%) !default;
...
-        font-weight: $slider__price-new-font-weight;
-    }
-
-    &__badge {
-        position: absolute;
-        top: $slider__badge-top;
-        left: 0;
-    }
-
     &__navigation {
```
Comment (anqaka) [inline components/03-modules/slider/_slider.scss]: It's a problem with `slider__item` height. On FF and Safari it seems ok but on Chrome the width is much bigger than the image (I put the image from banner and test different configuration). The `height: fit-content` or `height: max-content` for  `slider__item` fix the issue but it not [fully supported](https://caniuse.com/#search=fit-content) so it must be tested on mobiles.

File: components/03-modules/slider/slider.js [hunk 3]
```diff
@@ -6,13 +6,19 @@
   const sliders = [...document.querySelectorAll('.slider')];
 
   sliders.forEach(slider => {
-    lory(slider, {
-      rewind: true,
-      enableMouseEvents: true,
+    const dataValues = slider.dataset;
```
Comment (anqaka) [inline components/03-modules/slider/slider.js]: I found other bug, I tried to use this code for main banner slider. I put the image form banner and test it, I changed the width of elem in scss to 100% and add attribute `data-slides-to-scroll="1"`, If I click on the `next` arrow, for the first time it goes one element right, if I click on that second time it goes two or three items right. Please check it.

File: components/03-modules/slider/_slider.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (kjugi) [reply components/03-modules/slider/_slider.scss]: ok, it's visible for now

File: components/03-modules/slider/_slider.scss [hunk 4]
```diff
@@ -63,83 +44,6 @@ $slider__icon-fill-hover           : darken($color-primary, 10%) !default;
         }
     }
```
[the remark this answers is not in the document]
Comment (kjugi) [reply components/03-modules/slider/_slider.scss]: b/c just `width` property doesn't work

File: components/03-modules/slider/slider.config.js [hunk 5]
```diff
@@ -17,88 +18,48 @@ module.exports = {
       iconClass: 'button__icon slider__icon',
       attributes: 'type="button" aria-label="button arrow slider next"'
     },
-    products: [
+    elements: [
```
[the remark this answers is not in the document]
Comment (kjugi) [reply components/03-modules/slider/slider.config.js]: ok

File: components/03-modules/slider/slider.js (hunk 3 above)
Comment (kjugi) [reply components/03-modules/slider/slider.js]: there was mistake with string typeof, for now it's parsing to int

File: components/03-modules/slider/_slider.scss (hunk 2 above)
Comment (kjugi) [reply components/03-modules/slider/_slider.scss]: error by type of variables (fixed for now) or error by lazyloading images and no placeholder for it

File: components/03-modules/slider/slider.config.js [hunk 6]
```diff
@@ -17,88 +18,48 @@ module.exports = {
...
-        heading: {
-          tag: 'h5',
-          class: 'slider__name',
-          text: 'Some product name',
-        },
-        price: '$ 200.00',
-        oldPrice: '$ 250.00'
+        content: 'image',
+        contentContext: {
+          src: '../../images/product/product-base.jpg'
```
[the remark this answers is not in the document]
Comment (kjugi) [reply components/03-modules/slider/slider.config.js]: changed with adding one `responsive-container` new global

File: components/03-modules/slider/slider.config.js [hunk 7]
```diff
@@ -1,6 +1,7 @@
 module.exports = {
   context: {
     script: true,
+    attributes: 'data-rewind="true" data-enable-mouse-events="true"',
```
[the remark this answers is not in the document]
Comment (kjugi) [reply components/03-modules/slider/slider.config.js]: `README.md` created

File: components/03-modules/slider/_slider.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (kjugi) [reply components/03-modules/slider/_slider.scss]: `min-width` are in variables now

**Comment by kjugi** [comment]
Ok, some files changed so let me explain.

For now, we don't keep lazyloaded images, it's the only preview and this issue is not related to the slider. There was created the other one for this #215 . I've deleted code with aspect-ratio b/c it's not existing in develop for now and required some fixes before we can use it. Now we have a working slider and it's ok.

**Comment by anqaka** [comment]
waiting for working lazyloading images inside the slider

**Comment by kjugi** [comment]
as we discussed on meeting, this changes should be in develop for now

**Pull request merged**